### ISSUE TITLE
Cannot import from previous export

### ISSUE THREAD

**Issue by steveman1123** [TeamNewPipe/NewPipe#7802]
Steps to reproduce:  
1. Under subscriptions tab, select subscritions item.  
2. Select "import from"  "previous export"
3. select the exported json file
4. subscritions are populated

Expected behavior:
Subscriptions are populated from the json export

Actual behavior:
No files are able to be selected in the file browser, all files are greyed out. 

Additional steps taken:
Storage permission is allowed, import also works in newpipe legacy v0.20.8

Android version:
8.0.0

Newpipe version:
0.21.15/0.21.16

**Comment by steveman1123** [comment]
No template was generated when writing this issue (could be due to using mobile site), apologies for any inconvience. Please comment for any additional info as this is my first issue with newpipe

**Comment by mhmdanas** [comment]
> No template was generated when writing this issue (could be due to using mobile site), apologies for any inconvience.

No worries, this is already better than a lot of issues out there.

I can't reproduce this. What file manager are you using? Can you try https://f-droid.org/packages/me.zhanghai.android.files and see if it works?

**Comment by steveman1123** [comment]
I don't seem to have a choice, it just automatically opens the default android file browser (it looks like the legacy one opens its own browser)

**Comment by mhmdanas** [comment]
> I don't seem to have a choice, it just automatically opens the default android file browser

Did you try installing it yet or not?

> it looks like the legacy one opens its own browser

The current NewPipe still has that one, but it's used only when SAF is off. I think on Android 8 we enabled it by default at some point. I think you can disable SAF by toggling "Settings > Download >  Use system folder picker (SAF)" off, but I'm not sure since I'm using Android 11, where SAF is mandatory.

**Comment by steveman1123** [comment]
That did it! Updated perfectly after disabling SAF

**Comment by mhmdanas** [comment]
That's a workaround, not a fix, so I'll re-open this.

**Comment by mhmdanas** [comment]
>  Importing a zip from the subscriptions tab doesn't work with neither of my file managers.

That's working as intended, as the subscription file format is JSON, not ZIP.

> In Settings -> Content, I can only select the exported zip with the Google file manager as selecting it with Material Files will result in "No valid ZIP file" error and even then, it will only reload the app without changing anything.

Maybe the ZIP file was corrupted? What version of NewPipe did you export the ZIP from? Or was it exported from a debug version of a PR or something?

**Comment by mrflamist** [comment]
> That's working as intended, as the subscription file format is JSON, not ZIP.

Aha, apologies. My bad.

> Maybe the ZIP file was corrupted? What version of NewPipe did you export the ZIP from? Or was it exported from a debug version of a PR or something?

I exported it on 0.21.15 from F-Droid. I may have just done something wrong.

**Comment by mhmdanas** [comment]
> I exported it on 0.21.15 from F-Droid. I may have just done something wrong.

It doesn't sound like it, but I'm not sure. Can you record your screen while you're trying to import the database but then NewPipe restarts without doing anything?

**Comment by mrflamist** [comment]
> It doesn't sound like it, but I'm not sure. Can you record your screen while you're trying to import the database but then NewPipe restarts without doing anything?

I'm afraid I cannot, because I just tried it one more time and it imported the database without a problem. Unfortunately, I have no idea what I did to make it work

**Comment by mhmdanas** [comment]
No problem, at least the issue is solved.

**Comment by NewNewPipe** [comment]
The application could add its version number to the export file which would make it easier to identify the source of the problem (between different versions).

**Comment by theduckrules** [comment]
When I export my YouTube subscriptions the only format available for subscriptions is .csv.  There is no option to export them as json.

**Comment by theduckrules** [comment]
Sorry. I thought this was a discussion of importing subscriptions

**Comment by codingForCats** [comment]
I have this problem. I can import a json file that I export from newpipe, but it greys out any json file I've emailed from freetube

**Comment by osopolar** [comment]
**tl:dr** It seems that you need to download the zip archive to your local storage, i.e. Download folder.

----

I migrated my old phone to a new using androids built-in transfer tool. It did not transfer new pipe data (subscriptions, playlists etc.). So I exported the data on the old phone, stored it in google drive, as it was the easiest way to get it to the new phone, and tried to import it from there – directly from the Google Drive folder. I got the "**No valid ZIP file**" error.

[Above comment](https://github.com/TeamNewPipe/NewPipe/issues/7802#issuecomment-1029871350) from, @mhmdanas  about "Use system folder picker (SAF)" helped me. In my settings it's greyed out and says:
> Starting from Android 10 only 'Storage Access Framework' is supported

On the new phone I downloaded the zip archive from my drive folder to the download folder (local storage) and then imported from there. Now it worked.

**Comment by DavidPesticcio** [comment]
I also cannot import previously exported subscriptions from the same directory as the export or any other. Restarting device has no effect. :disappointed: 

* NewPipe Version: 0.25.2
* Tablet Device: Amazon Fire 7
* Android OS: Fire OS 5.7.0.0

Works fine for the same version on Pixel 6 Pro running Android 12 - both installed from F-Droid.

**Comment by sudoshindo** [comment]
What's the status on this?

As of this comment, I'm on the latest version. I still cannot use this import function properly.

If there are no plans to fix this then can you instead just bind these items to the Import/Export Database functions found in the settings?

**Comment by sudoshindo** [comment]
> Tried this? [#7802 (comment)](https://github.com/TeamNewPipe/NewPipe/issues/7802#issuecomment-1435771725)

Completely irrelevant. I'm not sharing the exported file with Drive.

**Comment by fearedbliss** [comment]
Hello, I've also been experiencing this issue for a while, but finally got around to filing a bug report and found this one. Below is an export of my subscriber list, we can see that the issue is that the exporter is inserting extra characters in the middle of two json elements, causing the entire export to fail to be read. I haven't looked at the code but it may be around some sort of iteration logic. The below was exported using the built in NewPipe exporting functionality:

```
{"app_version":"0.27.2","app_version_int":999,"subscriptions":[{"service_id":0,"url":"https://www.youtube.com/channel/UCRtAu8OVYVfuNZmTV-Ug0zA","name":"Buddhism"},{"service_id":0,"url":"https://www.youtube.com/channel/UCPIyEJzvW7SsbiIrooixjNA","name":"Doug's Dharma"},{"service_id":0,"url":"https://www.youtube.com/channel/UCDiKw3GnFIwyNJBzhCoRI-Q","name":"Framework"},{"service_id":0,"url":"https://www.youtube.com/channel/UCRIeMHsEdzA9RroG19kXdYg","name":"Jesse Enkamp"},{"service_id":0,"url":"https://www.youtube.com/channel/UCy7jKqbTj9DqgIeaQT0_png","name":"Kevin Lee"},{"service_id":0,"url":"https://www.youtube.com/channel/UC5UAwBUum7CPN5buc-_N1Fw","name":"The Linux Experiment"}]},{"service_id":0,"url":"https://www.youtube.com/channel/UCy7jKqbTj9DqgIeaQT0_png","name":"Kevin Lee"},{"service_id":0,"url":"https://www.youtube.com/channel/UChJQlg1b_cOttPX3SiIh5gA","name":"Lau Dinha in Hong Kong - Hong Kong in the World"},{"service_id":0,"url":"https://www.youtube.com/channel/UCXXd3koJ2dagmhR1csXRTfw","name":"Moo-Moo Clips"},{"service_id":0,"url":"https://www.youtube.com/channel/UCXYOXWWqgtSrk27402V_JMw","name":"Sensei Seth"},{"service_id":0,"url":"https://www.youtube.com/channel/UCU4WVZqK_vuXb6sAArbs8yQ","name":"Supper Moment"},{"service_id":0,"url":"https://www.youtube.com/channel/UC5UAwBUum7CPN5buc-_N1Fw","name":"The Linux Experiment"}]}
```

And we can see the message that was shown by the [DuckDuckGo Json Validator](https://duckduckgo.com/?t=h_&q=json+validator&ia=answer):

```
Error: Parse error on line 1:
...Linux Experiment"}]},{"service_id":0,"ur
-----------------------^
Expecting 'EOF', got ','
```

**Comment by Stypox** [comment]
@fearedbliss I have no idea what might have happened to the export you made, but it seems likely that the file got corrupted in some way, because the file is valid JSON up until the comma that gives the error, and the rest of the file contains duplicate channels. Can you reproduce consistently, or did this happen to just one file?

**Comment by fearedbliss** [comment]
It may be possible it happened due to some weird interactions with the workflow I described above. I tried to export/import by creating a new file (rather than overwriting an existing file) and that seemed to work correctly. Feel free to consider what I said before as either invalid or potentially a minor bug that can be deprioritized).

**Comment by fearedbliss** [comment]
Thanks for the fix @Stypox. I'm happy you were able to reproduce and it wasn't just me going crazy haha.

**Comment by rkrisztian** [comment]
I could import Subscriptions from NewPipe JSON format like this on GrapheneOS (Pixel 9):

* Compress the NewPipe JSON into a ZIP file. (I could do this on the phone with the _Compress_ function of the built-in Android file manager.)
* Subscriptions -> Import from -> Previous Export
* Select the ZIP file created in the first point.

I don't install any unnecessary apps if I can afford, so at least this workaround worked for me.

**Comment by CMoH** [comment]
In my case it was a file permission error: I had copied my file onto the LineageOS RPI4 SD card on my PC, so the file and folder had owner root:1023 and the NewPipe export zip had rw-r--r--. This brought the "no valid zip file" message. Following the example of a test export file I changed the file owner on PC to 10063:1023, and rw-rw----, and then NewPipe was able to import the data successfully. I thought this report might help others.

As a suggestion, it might be useful to check file readability and explicitly report such failures. I browsed for about an hour, and finally an unrelated note about permissions earlier in this discussion barely made me think to try this, as a shot in the dark.

**Comment by fgaz** [comment]
Sailfish OS Android AppSupport here, also a permission error.
I downloaded the file from the native Sailfish side, so it had the wrong owner for AppSupport.

To fix the issue, I obtained the uid of other AppSupport files by running `ls -ld /home/defaultuser/android_storage/Download` and ran `chown <uid from previous step> export.zip`.
